src/VisualStudio/CSharp/Impl/Snippets/SnippetCommandHandler.cs

```diff
@@ -73,11 +85,29 @@ public CommandState GetCommandState(SurroundWithCommandArgs args)
             return CommandState.Available;
         }
 
+        public CommandState GetCommandState(TypeCharCommandArgs args, Func<CommandState> nextCommandHandler)
+        {
+            return nextCommandHandler();
+        }
+
+        public void ExecuteCommand(TypeCharCommandArgs args, Action nextCommandHandler, CommandExecutionContext executionContext)
+        {
+            AssertIsForeground();
+            if (args.TypedChar == ';'
+                && AreSnippetsEnabled(args)
+                && args.TextView.Properties.TryGetProperty(typeof(AbstractSnippetExpansionClient), out AbstractSnippetExpansionClient snippetExpansionClient))
+            {
+                snippetExpansionClient.TryHandleReturn();
```

Comment:
i get what you're doing, but this is worth an explanation in code for the next person who comes along :)  i.e. why is it that `;` is being treated as 'return'.

Reply:
⏱️ Extracted a helper method `CommitSnippet`. `TryHandleReturn` now calls this helper, and the new calls invoke `CommitSnippet` directly instead of going through `TryHandleReturn`.